src/esmac_diags/preprocessing/prep_ACEENA_E3SM.py

```diff
@@ -1,1286 +1,1442 @@
+        mom_all.coords['time'] = mom_all.indexes['time'].to_datetimeindex()
+        ncl_all.coords['time'] = ncl_all.indexes['time'].to_datetimeindex()
+        pom_all.coords['time'] = pom_all.indexes['time'].to_datetimeindex()
+        so4_all.coords['time'] = so4_all.indexes['time'].to_datetimeindex()
+        soa_all.coords['time'] = soa_all.indexes['time'].to_datetimeindex()
+    else:
+        bc_all  = xr.DataArray(np.zeros(len(e3smtime))*np.nan,attrs={'units':'dummy_unit','long_name':'Dummy'})
+        dst_all = xr.DataArray(np.zeros(len(e3smtime))*np.nan,attrs={'units':'dummy_unit','long_name':'Dummy'})
+        mom_all = xr.DataArray(np.zeros(len(e3smtime))*np.nan,attrs={'units':'dummy_unit','long_name':'Dummy'})
+        ncl_all = xr.DataArray(np.zeros(len(e3smtime))*np.nan,attrs={'units':'dummy_unit','long_name':'Dummy'})
+        pom_all = xr.DataArray(np.zeros(len(e3smtime))*np.nan,attrs={'units':'dummy_unit','long_name':'Dummy'})
+        so4_all = xr.DataArray(np.zeros(len(e3smtime))*np.nan,attrs={'units':'dummy_unit','long_name':'Dummy'})
+        soa_all = xr.DataArray(np.zeros(len(e3smtime))*np.nan,attrs={'units':'dummy_unit','long_name':'Dummy'})
+    
+    # aerosol size
+    req_vlist = ['num_a1', 'num_a2', 'num_a3', 'num_a4', 'dgnd_a01', 'dgnd_a02', \
+                 'dgnd_a03', 'dgnd_a04']
+    req_vlist = ["{}_{}".format(i,E3SMdomain_range) for i in req_vlist]
+    matched_vlist = list(set(av_vars).intersection(req_vlist))
+    
+    if len(matched_vlist) == len(req_vlist):
+        num_a1 = e3smdata['num_a1'+'_'+E3SMdomain_range].load()
+        num_a2 = e3smdata['num_a2'+'_'+E3SMdomain_range].load()
+        num_a3 = e3smdata['num_a3'+'_'+E3SMdomain_range].load()
+        num_a4 = e3smdata['num_a4'+'_'+E3SMdomain_range].load()
+        dn1 = e3smdata['dgnd_a01'+'_'+E3SMdomain_range].load()
+        dn2 = e3smdata['dgnd_a02'+'_'+E3SMdomain_range].load()
+        dn3 = e3smdata['dgnd_a03'+'_'+E3SMdomain_range].load()
+        dn4 = e3smdata['dgnd_a04'+'_'+E3SMdomain_range].load()
+        Pres = np.nan*T
+        zlen = T.shape[1]
+        for kk in range(zlen):
+            Pres[:, kk, :] = hyam[kk]*P0  +  hybm[kk]*PS
+        numall = [num_a1[:, -1, x_idx].data, num_a2[:, -1, x_idx].data, num_a3[:, -1, x_idx].data, num_a4[:, -1, x_idx].data]
+        dnall  = [dn1[:, -1, x_idx].data,    dn2[:, -1, x_idx].data,    dn3[:, -1, x_idx].data,    dn4[:, -1, x_idx].data]
+        NCN = calc_CNsize_cutoff_0_3000nm(dnall, numall, T[:, -1, x_idx].data, Pres[:, -1, x_idx].data)
+        NCNall = xr.DataArray(data=NCN,  dims=["size", "time"],
+            coords=dict(time=(["time"], e3smtime),size=(["size"], np.arange(1,3001))),
+            attrs=dict(long_name="Aerosol number size distribution",units="#/m3"),)
+    else:
+        NCNall = xr.DataArray(np.zeros((3000,len(e3smtime)))*np.nan,attrs={'units':'dummy_unit','long_name':'Dummy'})
+    
+    # variables to calculate Reff and Nd
```

Comment:
this part should be "cloud heights and depth"

Reply:
Changed the comments for all 4 files.